### PR TITLE
[4.0] Update Tinymce

### PR DESCRIPTION
**Pull request by brianteeman**
TinyMce follows semantic versioning. There is no reason that we should be locked down to an old patch release of 5.0.14 when the current is 5.1.5

~Some help fixing the javascript in build\build-modules-js\init.es6.js to write the correct version number in the plugin xml would be appreciated~

### CODE REVIEW

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on b620847b13c0bcd4d6843434b17a8005db5e2fa6<br /><br />Tested all functions of Tinymce on backend and frontend. Everything works as intended.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27520">issues.joomla.org/tracker/joomla-cms/27520</a>.</sub>

File: package.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
     "qrcode-generator": "^1.4.3",
     "roboto-fontface": "^0.10.0",
     "skipto": "^2.0.4",
-    "tinymce": "5.0.14"
+    "tinymce": "^5.1.0"
```
Comment (Quy) [inline package.json:52]: Should this be `5.1.5`?

Comment (brianteeman) [reply package.json:52]: no it's fine like this - as you see in the lock file it is 5.1.5 thats installed. 
we dont need to specify every patch

File: plugins/editors/tinymce/tinymce.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <extension version="3.2" type="plugin" group="editors" method="upgrade">
 	<name>plg_editors_tinymce</name>
-	<version>5.0.14</version>
+	<version>^5.1.0</version>
```
Comment (mbabker) [inline plugins/editors/tinymce/tinymce.xml]: Apparently the reason it's pinned to a specific version is right here...

Comment (brianteeman) [reply plugins/editors/tinymce/tinymce.xml]: Which is stupid

Comment (SharkyKZ) [reply plugins/editors/tinymce/tinymce.xml]: PR https://github.com/brianteeman/joomla-cms/pull/90 should fix this.

Comment (SharkyKZ) [reply plugins/editors/tinymce/tinymce.xml]: PR https://github.com/brianteeman/joomla-cms/pull/90 should fix this.

**Comment by brianteeman** [comment]
Thanks. I will test and merge in the morning.

**Comment by brianteeman** [comment]
@SharkyKZ I applied your pr and ran npm i
but the version in the xml is still
`	<version>^5.1.0</version>`
did i miss something?

**Comment by SharkyKZ** [comment]
It's because regex for replacing version expects a valid version in XML. As a quick fix you can manually change the version just in the XML to something valid (e.g. back to 5.0.14) and then run `npm install` again.

**Comment by SharkyKZ** [comment]
Although I don't see the need for such specific regex. We could simplify it. See https://github.com/brianteeman/joomla-cms/pull/91.

**Comment by brianteeman** [comment]
Thanks @SharkyKZ the regex works perfectly now.

**Comment by jwaisner** [comment]
@brianteeman During testing everything works as expected, just need to confirm that the version presented in the extension manager will still be 5.0.14 or should that show 5.1.5?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27520">issues.joomla.org/tracker/joomla-cms/27520</a>.</sub>

**Comment by SharkyKZ** [comment]
It will probably show 5.0.14. You can use `Refresh Cache` button to update data from manifest.

@brianteeman You still need to commit XML changes to correct version.

**Comment by jwaisner** [comment]
Refreshing the Cache worked. I noticed the version shows as "^5.1.0". Is that supposed to have the "^" in the version number? Changing `<version>^5.1.0</version>`  to `<version>5.1.0</version>` corrects this. I dont fully understand the versioning in Tinymce, so this may be normal.

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on ed913983f1289c5307d99e616b0c6484d214e0fb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27520">issues.joomla.org/tracker/joomla-cms/27520</a>.</sub>

**Comment by brianteeman** [comment]
Can someone please check and merge this as it addresses a security vulnerability

**Comment by SharkyKZ** [comment]
You still need to run `npm install` to update the XML with correct version.

**Comment by brianteeman** [comment]
@SharkyKZ doh - done now

**Comment by brianteeman** [comment]
Can someone please check and merge this as it addresses a security vulnerability

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**